.github/actions/build_and_publish_image/action.yaml

```diff
@@ -0,0 +1,68 @@
+name: Build & publish an image
+
+description: Build an image from source, then publish it to GHCR
+
+inputs:
+  build_args:
+    description: '`ARG=value`s to pass to `docker build`'
+    required: false
+  dockerfile:
+    description: Image source file name
+    default: 'Dockerfile'
+    required: false
+  environment:
+    description: The environment for the image, such as "dev" or "master"
+    required: true
+  image_source_directory:
+    description: Image source directory path
+    required: true
+  image_name:
+    description: The basename for the image tag, such as "image" in "ghcr.io/user/repo/image:latest"
+    required: true
+  image_version_1:
+    description: A version for the image tag, such as "foo" in "ghcr.io/user/repo/image:bar"
+    required: true
+  image_version_2:
+    description: Another version for the image tag, such as "bar" in "ghcr.io/user/repo/image:bar"
+    required: false
+  use_cache:
+    default: "true"
+    description: Use cached layers from pre-published image with version 1 (true/false)
+    required: false
+
+runs:
+  using: composite
+  steps:
+  - shell: bash
+    run: |
+      set -x
+
+      export TOKEN=${{ secrets.GITHUB_TOKEN }}
+      export USERNAME=${{ github.actor }}
+      echo $TOKEN | docker login ghcr.io -u $USERNAME --password-stdin
+      export BUILD_ARGS=""
+      if [[ -n "${{ inputs.build_args }}" ]]; then
+        IFS=";" read -ra args <<< "${{ inputs.build_args }}"
+        for arg in ${args[@]}; do BUILD_ARGS+="--build-arg $arg "; done
+      fi
+
+      export IMAGE_TAG_URL_PREFIX=ghcr.io/selkies-project/selkies-vdi/
+      if [[ "${{ inputs.environment }}" == "master" ]]; then
+        export IMAGE_TAG_ENVIRONMENT_SUFFIX=""
+      else
+        export IMAGE_TAG_ENVIRONMENT_SUFFIX="-dev"
```

Comment:
The only distinction between `dev` and `master` images is that `dev` image tags have a `-dev` suffix (ex: `ghcr.io/selkies-project/selkies-vdi/webrtc-gpu-streaming-app-streaming:latest-dev`

Reply:
This is a mistake - the release version is already the branch name.